andela/pippin-ah-backend
# [feature: #162727561]: In App Notification

#### What does this PR do?
- Handle inApp notification

#### Description of Task to be completed?
- [x] install pusher
- [x] add client directory as a frontend template to test inapp functionality
- [x] create follow model and migration
- [x] create methods to send notification to followers when new articles are created
- [x] create method to send notification to authors when there is comment on their ariticle
- [x] create notification model and migration
- [x] create endpoint to get all notification
- [x] create endpoint to edit the status of notification from unread to read
- [x] modify register and login route to return all notifications that belong to a user
- [x] write test for the functionalities

#### How should this be manually tested? 
- clone the repo
- run `npm install` to install new packages (Pusher) 
- run `npm install db:migrate` to create the new database
- Make sure; at least two users are created, a user has article in database and has atleas a follower
- fire up postmant
- navigate to PATCH `localhost:3000/api/v1/user/notifications?notificationId=<id of notification from notification table>` to change notification status from `unread` to `read`
- navigate to  GET `localhost:3000/api/v1/user/notifications` to get all notifications belonging to the user

#### What are the relevant pivotal tracker stories?
[#162727561](https://www.pivotaltracker.com/story/show/162727561)
